src/tscore/CMakeLists.txt

```diff
@@ -103,6 +102,14 @@ add_library(tscore SHARED
 )
 add_library(ts::tscore ALIAS tscore)
 
+if(OPENSSL_IS_BORINGSSL)
+    target_sources(tscore PUBLIC HKDF_boringssl.cc)
+elseif(OPENSSL_IS_OPENSSL3)
+    target_sources(tscore PUBLIC HKDF_openssl3.cc)
+else()
+    target_sources(tscore PUBLIC HKDF_openssl.cc)
```

Comment:
I couldn't figure out which was the default for add_library so just punted.  I think we need to audit these usage requirements in general.

Reply:
We do. I've been fixing a few here and there, but most of it will probably have to be part of the general cleanup. Regardless, we should make sure that new ones are correct.

Part of the general problem there is that the cyclic dependencies result in usage requirements being weird.